src/app/perlenkette/service/load-perlenkette.service.ts

```diff
@@ -123,6 +123,12 @@ export class LoadPerlenketteService implements OnDestroy {
     while (alltrainrunsections.length > 0) {
       // traverse over all trainrun parts
       const trainrunSection = alltrainrunsections[0];
+
+      // filter all still visited trainrun sections
+      alltrainrunsections = alltrainrunsections.filter(ts =>
```

Comment:
```suggestion
      allTrainrunSections = allTrainrunSections.filter(ts =>
```

could you rename the upper occurrences as well ?